### PR TITLE
Fix film for discrete knoks

### PR DESCRIPTION
**Pull request by vallsv**
Hi,

Here is a fix to use the full available film in case of discrete knoks.

That is important cause i have to align this knok with labels for each discrete values.

Is this code also used for the Mod Duo server? Or it have to be backported?

Here is an example with a knok with 6 values (integer from 0 to 5 included). On the left the `master`, on the right the fix.
fix-discrite-knoks

The code still looks weird cause a double round sounds like it is possible to accumulate many delta error. But i mostly understand nothing to the code (what is `steps` and so on).

### CODE REVIEW

**Comment by falkTX** [comment]
That makes a lot of sense actually. Will merge, thanks.

btw, it is written *knob*, not knok

**Comment by falkTX** [comment]
we can do the same fix in mod-ui, that is the webserver inside the MOD Duo units.

**Comment by vallsv** [comment]
Thanks a lot.
And I am kind of dyslexial. I had to fix many times this typo on by code :-)

**Comment by leogermani** [comment]
Hi guys,

Its been a while since I was working on this so I can't tell it for myself. But could you have a look at this documentation and check whether there is something that should be changed?

https://wiki.moddevices.com/wiki/Preparing_the_assets#Knobs.2C_sliders_and_similars

See that in the docs I suggest to prepare the sprite image only with the steps the knob will have so you get full control to align them. 

In other words, in those cases you dont let the UI auto-calculate the rotation, you prepare the image with each step pointing to the right place (It seems to me a better idea).

**Comment by vallsv** [comment]
The documentation is still fine.

But i think in most of the cases it is easier to use the continuous sprite to deal with discrete cases. Preparing sprites is a lot of time, and if i have to change the style, it's much easier.

**Comment by blushi** [comment]
That seems fine to me, it removes the "undesired behavior" mentioned by the wiki.
